sample/marchingSquares.glsl

```diff
@@ -0,0 +1,204 @@
+#include "../generative/pnoise.glsl"
+#include "../math/const.glsl"
+#include "../math/rotate2d.glsl"
+#include "../sample.glsl"
+
+/*
+Author: Guido Schmidt
+description: The Marching Squares algorithm generates contour lines (isolines) of a given input for a
+given grid and a threshhold. Is uses a lookup of 16 cases to decide if a cell is inside, outside or
+a specific corner.
+use: <vec3> marchingSquares(in <vec2> uv) 
+options:
+    - SAMPLE_MARCHING_SQUARES_FNC(UV): optional sampling function, defaults so sampling of pnoise
+*/
+
+#ifndef SAMPLE_MARCHING_SQUARES_FNC
```

Comment:
the name of this macros should be

```
SAMPLEMARCHINGSQUARES_FNC(TEX, UV)
```

I see no sign of texture sample... would be that possible to add?
We could have to main sampleMarchingSquares functions one that takes a texture and one that doesn't

Reply:
Oh yes, I wanted to add the possibility to sample from a texture